pilot/test/integration/egress_rules_http_tls_origination.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Routing tests
+
+package main
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	multierror "github.com/hashicorp/go-multierror"
+)
+
+type egressRulesHTTPTLSOrigination struct {
+	*egressRules
+}
+
+func (t *egressRulesHTTPTLSOrigination) String() string {
+	return "egress-rules-http-tls-origination"
+}
+
+// TODO: test negatives
+func (t *egressRulesHTTPTLSOrigination) run() error {
+	cases := []struct {
+		description string
+		config      string
+		check       func() error
+	}{
+		{
+			description: "allow https external traffic to *google.com",
+			config:      "egress-rule-google.yaml.tmpl",
+			check: func() error {
+				return t.verifyReachable("http://cloud.google.com:443", true)
```

Comment:
Let’s please not do this. We should just have https and ensure that we have a tcp listener on 443

Reply:
OR a http://cloud.google.com/foo and check if it still works.

Reply:
But we will still allow tls origination by proxy, won't we? So we will have to test it.

Reply:
Yes. But we know that cloud.googlr.com will not accept plain http. So even if we access the site via http, proxy should upgrade to https (if the tls origin flag was set in the egress rule). Make sense?

This can be done in separate pr

Reply:
Agreed to be done in a separate PR.